tests/subsys/settings_commit_prio/src/main.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2024 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/settings/settings.h>
+#include <zephyr/ztest.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(test);
+
+static int prio;
+
+int commit0(void)
+{
+	LOG_INF("%s Called", __func__);
+	if (prio != 0) {
+		return -EINVAL;
+	}
```

Comment:
Why is this not
```suggestion
	zassert_equal(prio, 0);
```

Reply:
Returning `-EINVAL` allows to check that the result of `commit` is 0.

Reply:
Can you elaborate? If we know it's going to fail here, why not assert here? Earlier terminator, and I suppose it will be clearer where it fails?